assets/src/edit-story/components/panels/panel/panel.js

```diff
@@ -20,37 +20,93 @@
 import styled from 'styled-components';
 import PropTypes from 'prop-types';
 import { v4 as uuidv4 } from 'uuid';
-import { useState, useCallback } from 'react';
+import { useState, useCallback, useEffect } from 'react';
 
 /**
  * Internal dependencies
  */
 import panelContext from './context';
 
+export const PANEL_COLLAPSED_THRESHOLD = 10;
+
 const Wrapper = styled.section`
   display: flex;
   flex-direction: column;
 `;
 
-function Panel({ initialHeight, name, children }) {
+function Panel({ resizeable, initialHeight, name, children }) {
   const [isCollapsed, setIsCollapsed] = useState(false);
+  const [expandToHeight, setExpandToHeight] = useState(initialHeight);
   const [height, setHeight] = useState(initialHeight);
+  const [manuallyChanged, setManuallyChanged] = useState(false);
+
+  const collapse = useCallback(() => {
+    setIsCollapsed(true);
+    if (resizeable) {
+      setHeight(0);
+    }
+  }, [resizeable]);
+  const expand = useCallback(
+    (restoreHeight = true) => {
+      setIsCollapsed(false);
+      if (restoreHeight && resizeable) {
+        setHeight(expandToHeight);
+      }
+    },
+    [resizeable, expandToHeight]
+  );
+
+  useEffect(() => {
+    if (
+      resizeable &&
+      height <= PANEL_COLLAPSED_THRESHOLD &&
+      isCollapsed === false
+    ) {
+      collapse();
+    }
+  }, [collapse, height, resizeable, isCollapsed]);
+
+  useEffect(() => {
+    if (manuallyChanged || !resizeable) {
+      return;
+    }
+    setHeight(initialHeight);
+    setExpandToHeight(initialHeight);
+  }, [manuallyChanged, initialHeight, resizeable]);
+
+  const manuallySetHeight = useCallback(
+    (h) => {
+      if (!resizeable) {
+        return;
+      }
+      setManuallyChanged(true);
+      setHeight(h);
+      if (isCollapsed && h(height) > PANEL_COLLAPSED_THRESHOLD) {
+        expand(false);
+      }
+    },
+    [setManuallyChanged, setHeight, height, expand, resizeable, isCollapsed]
+  );
 
-  const collapse = useCallback(() => setIsCollapsed(true), []);
-  const expand = useCallback(() => setIsCollapsed(false), []);
+  const resetHeight = useCallback(() => {
+    setManuallyChanged(false);
+  }, []);
```

Comment:
If the panel is collapsed and I double-click the dragbar, it'll reset the height, but not actually expand the panel. Shouldn't it?

Reply:
Added